### PR TITLE
httpbakery/client: catch non json discharge responses

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

### CODE REVIEW

File: macaroonbakery/tests/test_client.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import base64
 import datetime
 import json
+from json.decoder import JSONDecodeError
```
Comment (mhilton) [inline macaroonbakery/tests/test_client.py:6]: travis is complaining about this line.

**Review by mhilton — APPROVED** [review]
This seems fine to me once travis is happy, but I'm going to seek a second opinion.

File: macaroonbakery/httpbakery/_client.py [hunk 2]
```diff
@@ -132,6 +133,13 @@ def acquire_discharge(self, cav, payload):
         if resp.status_code == 200:
             return bakery.Macaroon.from_dict(resp.json().get('Macaroon'))
         cause = Error.from_dict(resp.json())
```
Comment (frankban) [inline macaroonbakery/httpbakery/_client.py:135]: Is this line a leftover?
Assuming this raises a JSONDecodeError, how is it possible that the test is passing?

**Review by frankban — COMMENTED** [review]
Looks great with a question. Thanks!

File: macaroonbakery/httpbakery/_client.py (hunk 2 above)
Comment (sergiusens) [reply macaroonbakery/httpbakery/_client.py:135]: yeah, sorry, I was making sure my unit test worked and then failed to remove it

File: macaroonbakery/tests/test_client.py (hunk 1 above)
Comment (sergiusens) [reply macaroonbakery/tests/test_client.py:6]: I will change it to a ValueError instead which is what the JSONDecodeError is based out of.

**Review by fabricematrat — APPROVED** [review]

**Pull request merged**